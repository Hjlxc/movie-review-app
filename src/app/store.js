import { configureStore } from '@reduxjs/toolkit';
import movieDataReducer from '../modules/movieData';

export default configureStore({
  reducer: {
    movieData: movieDataReducer,
  },
});
