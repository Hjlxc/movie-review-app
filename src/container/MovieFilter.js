import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Divider } from 'antd';

import { VerticalFlexWrapper } from './styledComponent';
import {
  setMovieFilterVoting,
  setMovieFilterLanguage,
  setMovieFilterAdult,
  setMovieFilterSearch,
  setMovieSort,
  selectMovieVoting,
  selectLanguageOption,
  selectMovieLanguage,
  selectMovieAdult,
  selectMovieSort,
  sortOptions,
} from '../modules/movieFilter';
import {
  CheckboxGroup as LanguageFilter,
  Slider as VotingFilter,
  Switch as AdultFilter,
  SearchBox as SearchFilter,
  Dropdown as SortDropdown,
} from '../component';
import TitleWrapper from './TitleWrapper';

const sortDropdownOptions = Object.keys(sortOptions);

export default function MovieFilter() {
  const dispatch = useDispatch();

  // read filter data from redux store
  const voting = useSelector(selectMovieVoting);
  const languageOptions = useSelector(selectLanguageOption);
  const language = useSelector(selectMovieLanguage);
  const adult = useSelector(selectMovieAdult);
  const sort = useSelector(selectMovieSort);

  // update redux store when search compoennt updates
  const onVotingFilterChange = ([min, max]) =>
    dispatch(setMovieFilterVoting({ min, max }));

  const onLanguageFilterChange = (option) =>
    dispatch(setMovieFilterLanguage(option));

  const onAdultFilterChange = (check) => dispatch(setMovieFilterAdult(check));

  const onSearchFilterChange = (search) => {
    dispatch(setMovieFilterSearch(search));
  };

  const onSortFilterChange = (value) => {
    dispatch(setMovieSort(value.key === 'item_0' ? '' : value.key));
  };

  return (
    <VerticalFlexWrapper>
      <SearchFilter
        placeholder="Search Movie by Title"
        onSearch={onSearchFilterChange}
      />
      <Divider />
      <TitleWrapper title="Language">
        <LanguageFilter
          options={languageOptions}
          checked={language}
          onOptionClick={onLanguageFilterChange}
        />
      </TitleWrapper>
      <Divider />
      <TitleWrapper title="Adult Only" align="horizontal">
        <AdultFilter checked={adult} onChange={onAdultFilterChange} />
      </TitleWrapper>
      <Divider />
      <TitleWrapper title="Voting">
        <VotingFilter {...voting} onChange={onVotingFilterChange} />
      </TitleWrapper>
      <Divider />
      <TitleWrapper title="Sort By:" align="horizontal">
        <SortDropdown
          select={sort}
          options={sortDropdownOptions}
          onChange={onSortFilterChange}
        />
      </TitleWrapper>
    </VerticalFlexWrapper>
  );
}
