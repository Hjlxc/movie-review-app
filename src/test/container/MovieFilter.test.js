import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';

import MovieFilter from '../../container/MovieFilter';
import store from '../../app/store';

it('matches snapshot', () => {
  const testRenderer = renderer.create(
    <Provider store={store}>
      <MovieFilter />
    </Provider>
  );
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Test Redux connect', () => {
  let screen;

  beforeAll(() => {
    screen = render(
      <Provider store={store}>
        <MovieFilter />
      </Provider>
    );
  });

  it('Test child component rendered correctly', () => {
    expect(screen.getByPlaceholderText('Search Movie by Title')).not.toBeNull();
    expect(screen.getByText('Language')).not.toBeNull();
    expect(screen.getByText('Adult Only')).not.toBeNull();
    expect(screen.getByText('Sort By:')).not.toBeNull();
    expect(screen.getByText('Voting')).not.toBeNull();
  });
});
