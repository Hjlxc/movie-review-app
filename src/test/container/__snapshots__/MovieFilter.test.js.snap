// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxiKw jWoTjM"
>
  <span
    className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
  >
    <span
      className="ant-input-wrapper ant-input-group"
    >
      <input
        className="ant-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search Movie by Title"
        style={null}
        type="text"
        value=""
      />
      <span
        className="ant-input-group-addon"
      >
        <button
          className="ant-btn ant-input-search-button ant-btn-primary"
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
        >
          <span
            aria-label="search"
            className="anticon anticon-search"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </span>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="sc-AxiKw jWoTjM"
  >
    <div
      className="sc-AxhUy sc-AxheI bvzwzp"
    >
      Language
    </div>
    <div
      className="sc-AxiKw jWoTjM"
    />
  </div>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="sc-AxirZ hlnYRa"
  >
    <div
      className="sc-AxhUy sc-AxheI bvzwzp"
    >
      Adult Only
    </div>
    <div
      className="sc-AxgMl cEDzwU"
    />
    <button
      aria-checked={false}
      className="ant-switch"
      data-testid="switch"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="switch"
      type="button"
    >
      <div
        className="ant-switch-handle"
      />
      <span
        className="ant-switch-inner"
      />
    </button>
  </div>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="sc-AxiKw jWoTjM"
  >
    <div
      className="sc-AxhUy sc-AxheI bvzwzp"
    >
      Voting
    </div>
    <div
      className="ant-slider ant-slider-with-marks"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="ant-slider-rail"
        style={Object {}}
      />
      <div
        className="ant-slider-track ant-slider-track-1"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "100%",
          }
        }
      />
      <div
        className="ant-slider-step"
      >
        <span
          className="ant-slider-dot ant-slider-dot-active"
          style={
            Object {
              "left": "0%",
            }
          }
        />
        <span
          className="ant-slider-dot ant-slider-dot-active"
          style={
            Object {
              "left": "100%",
            }
          }
        />
      </div>
      <div
        aria-disabled={false}
        aria-valuemax={10}
        aria-valuemin={0}
        aria-valuenow={0}
        className="ant-slider-handle ant-slider-handle-1"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        aria-disabled={false}
        aria-valuemax={10}
        aria-valuemin={0}
        aria-valuenow={10}
        className="ant-slider-handle ant-slider-handle-2"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "100%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-slider-mark"
      >
        <span
          className="ant-slider-mark-text ant-slider-mark-text-active"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "left": "0%",
              "msTransform": "translateX(-50%)",
              "transform": "translateX(-50%)",
            }
          }
        >
          0
        </span>
        <span
          className="ant-slider-mark-text ant-slider-mark-text-active"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "left": "100%",
              "msTransform": "translateX(-50%)",
              "transform": "translateX(-50%)",
            }
          }
        >
          10
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="sc-AxirZ hlnYRa"
  >
    <div
      className="sc-AxhUy sc-AxheI bvzwzp"
    >
      Sort By:
    </div>
    <div
      className="sc-AxgMl cEDzwU"
    />
    <a
      className="ant-dropdown-link ant-dropdown-trigger"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
       
      <span
        aria-label="down"
        className="anticon anticon-down"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;
